subsys/net/lib/http/http_app_server.c

```diff
@@ -168,6 +168,7 @@ static void http_data_sent(struct net_app_ctx *app_ctx,
 int http_send_error(struct http_ctx *ctx, int code,
 		    u8_t *html_payload, size_t html_len)
 {
+	char buf[sizeof("HTTP/1.1 %3d\r\n\r\n")];
```

Comment:
This only works because `%3d` has the same number of characters as, say `200` or `404`. It might be a good idea to use `XXX` or something else here.

Reply:
It was done on purpose like this so that %3d is three letters wide. This way the same string can be used here and in snprintk(). Note that we only print the code if it is 100-999.